src/Schemes/Covering/Objects/Methods.jl

```diff
@@ -226,3 +226,84 @@ function Base.show(io::IO, C::Covering)
 #   print(io, glueing_graph(C))
 end
 
+########################################################################
+# Refinements                                                          #
+########################################################################
+
+@Markdown.doc """
+    common_refinement(C::Covering, D::Covering)
+
+For two `Covering`s `C` and `D`, calculate a common refinement 
+`E` and return a triple ``(E, φ, ψ)`` with ``φ : E → C`` 
+and ``ψ : E → D`` the `CoveringMorphism`s with the inclusion maps. 
+
+!!! note Since the `Covering`s do not know about any `AbsCoveredScheme`, 
+the computation of the refinement has to rely on the intrinsic tree
+structure of their `patches`. Due to these limitations, only special 
+cases are implemented; see the source code for details.
+"""
+function common_refinement(C::Covering, D::Covering)
+  # Check the easy cases: C is a refinement of D or the other way around.
+  if all(x->has_ancestor_in(patches(D), x), patches(C))
+    # C is a refinement of D
+    E = C
+    phi = identity_map(E)
+    map_dict = IdDict{AbsSpec, AbsSpecMor}()
+    for U in patches(C)
+      f, _ = _find_chart(U, D)
+      map_dict[U] = f
+    end
+    psi = CoveringMorphism(E, D, map_dict, check=false)
+    return E, phi, psi
+  elseif all(x->has_ancestor_in(patches(C), x), patches(D))
+    # D is a refinement of C
+    E = D
+    psi = identity_map(E)
+    map_dict = IdDict{AbsSpec, AbsSpecMor}()
+    for U in patches(E)
+      f, _ = _find_chart(U, C)
+      map_dict[U] = f
+    end
+    phi = CoveringMorphism(E, C, map_dict, check=false)
+    return E, phi, psi
+  end 
+  
+  error("case not implemented")
+  # We still need to adjust the code below.
```

Comment:
You are using error("not implemented") to comment out drafts (here and in at least one other place). Do you really want that draft code in the PR?

Reply:
Yes, why not? In this case, I started implementing a potentially horribly complicated function from which we only need some very special cases up to now. If your demand goes beyond the implemented levels, you are kindly informed about this. Precisely what we talked about two weeks ago and, I think, what even you suggested to do.